### PR TITLE
[Inductor] Match insignficiant strides on outputs

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122239

Fix for https://github.com/pytorch/pytorch/issues/116433

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

File: torch/_inductor/graph.py [hunk 1]
```diff
@@ -998,6 +1047,10 @@ def debug(msg):
                         stride_order = ir.NHWC_STRIDE_ORDER
                     result = ir.ExternKernel.require_stride_order(result, stride_order)
 
+                if is_output and len(strides):
```
Comment (shunting314) [inline torch/_inductor/graph.py:1050]: Does this need to be done for non user visible outputs?

File: torch/_inductor/graph.py [hunk 2]
```diff
@@ -920,6 +921,54 @@ def set_current_node(self, node: torch.fx.Node):
...
+            for dim, s1, s2 in zip(shape, meta_strides, tensor_strides):
+                if self.sizevars.statically_known_leq(dim, 1):  # type: ignore[arg-type]
+                    continue
+
+                if not self.sizevars.statically_known_equals(s1, s2):
+                    return False
+
+            return True
+
+        if not significant_strides_equal(
```
Comment (shunting314) [inline torch/_inductor/graph.py:952]: Get confused. Why we return directly here if significant strides are not equal?

File: torch/_inductor/graph.py (hunk 1 above)
Comment (eellison) [reply torch/_inductor/graph.py:1050]: Yes, because it interacts with aot_autograd

File: torch/_inductor/graph.py (hunk 2 above)
Comment (eellison) [reply torch/_inductor/graph.py:952]: We might have actual layout difference bc of things like channels last. In that case the `"did the strides diverge between inductor and eager"` is going to fire regardless of the size0/1 dimension stride checks

Comment (eellison) [reply torch/_inductor/graph.py:952]: I'll add comment

**Pull request closed without merging**